testsuite/MDAnalysisTests/analysis/test_gnm.py

```diff
@@ -26,126 +26,128 @@
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, -43.0, -3.0])
-
-    def test_closeContactGNMAnalysis_default_weights_is_size(self, universe):
-        gnm = MDAnalysis.analysis.gnm.closeContactGNMAnalysis(universe)
-        assert_equal(gnm.weights, "size")
-
-    def test_closeContactGNMAnalysis_deprecated_MassWeight_False(self, universe):
-        gnm = MDAnalysis.analysis.gnm.closeContactGNMAnalysis(universe, MassWeight=False)
-        assert_equal(gnm.weights, None)
-
-    def test_closeContactGNMAnalysis_deprecated_MassWeight_True(self, universe):
-        gnm = MDAnalysis.analysis.gnm.closeContactGNMAnalysis(universe, MassWeight=True)
-        assert_equal(gnm.weights, "size")
+@pytest.fixture()
+def universe():
+    return MDAnalysis.Universe(GRO, XTC)
+
+
+def test_gnm(universe, tmpdir):
+    output = os.path.join(str(tmpdir), 'output.txt')
+    gnm = MDAnalysis.analysis.gnm.GNMAnalysis(universe, ReportVector=output)
+    gnm.run()
+    result = gnm.results
+    assert len(result) == 10
+    time, eigenvalues, eigenvectors = zip(*result)
+    assert_almost_equal(time, np.arange(0, 1000, 100), decimal=4)
+    assert_almost_equal(eigenvalues,
+      [ 2.0287113e-15, 4.1471575e-15, 1.8539533e-15, 4.3810359e-15,
+        3.9607304e-15, 4.1289113e-15, 2.5501084e-15, 4.0498182e-15,
+        4.2058769e-15, 3.9839431e-15])
+
+
+def test_gnm_run_step(universe):
+    gnm = MDAnalysis.analysis.gnm.GNMAnalysis(universe)
+    gnm.run(step=3)
+    result = gnm.results
+    assert len(result) == 4
+    time, eigenvalues, eigenvectors = zip(*result)
+    assert_almost_equal(time, np.arange(0, 1200, 300), decimal=4)
+    assert_almost_equal(eigenvalues,
+      [ 2.0287113e-15, 4.3810359e-15, 2.5501084e-15, 3.9839431e-15])
+
+
+def test_generate_kirchoff(universe):
+    gnm = MDAnalysis.analysis.gnm.GNMAnalysis(universe)
+    gnm.run()
+    gen = gnm.generate_kirchoff()
+    assert_almost_equal(gen[0],
+      [ 7,-1,-1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0,-1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0,-1,-1,-1, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,-1, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
+
+
+def test_closeContactGNMAnalysis(universe):
+    gnm = MDAnalysis.analysis.gnm.closeContactGNMAnalysis(universe, weights="size")
+
+    gnm.run()
+
+    result = gnm.results
+    assert len(result) == 10
+    time, eigenvalues, eigenvectors = zip(*result)
```

Comment:
I don't think you are importing the iterating zip at the top. `from six.moves import zip` Please fix that for py2/3 compatibility. It might break the test then you need to write `list(zip(*result))`